tidy3d/components/heat_charge/monitor_data.py

```diff
@@ -181,6 +179,22 @@ def warn_no_data(cls, val, values):
 
         return val
 
+    @pd.validator("temperature", always=True)
+    @skip_if_fields_missing(["monitor"])
+    def check_correct_data_type(cls, val, values):
+        """Issue error if incorrect data type is used"""
+
+        mnt = values.get("monitor")
+
+        if isinstance(val, TetrahedralGridDataset) or isinstance(val, TriangularGridDataset):
+            if isinstance(val.values, IndexVoltageDataArray):
+                raise ValueError(
+                    f"Monitor {mnt} of type 'TemperatureMonitor' cannot be associated with data arrays "
+                    "of type 'IndexVoltageDataArray'."
+                )
```

Comment:
since we are likely to add others indexed data arrays, probably easier to switch here to something like:
```
if not isinstance(val.values, IndexDataArray):
    raise ValueError(
        f"Unstructured grid dataset given in 'TemperatureData.temperature' must contain only one field as 'IndexedDataArray'."
    )
```

Reply:
What happens if we have a `SpatialDataArray`? This would fail right? Since it doesn't have a `values` field?